### DISCUSSION TITLE
WRFHYDRO NOAHNWM-MP - description of input files

### DISCUSSION THREAD

**Discussion by jbensabat** [NCAR/wrf_hydro_nwm_public#721] [Q&A]
Hello 
I am new to WRFHYDRO. I am trying to construct a model 
I am starting with the preparation of input files with data collected locally and from LANDSAT and Copernicus. 
the first file is wrfinput_d01.nc 
Is there any written documentation about this file ? what are the mandatory variables and what are the optional ones ? what are the units etc .
thanks
jac

**Comment by scrasmussen** [comment]
Hi Jac,
Apologies for the slow response. I think looking at [section 5.2.2: Initial land surface conditions](https://wrf-hydro.readthedocs.io/en/latest/model-inputs-preproc.html#initial-land-surface-conditions) will help you do some of the things you are trying to do. Also, that first `wrfinput_d01.nc` file can also be produced with WRF's `real.exe` program so reading through some of [their documentation](https://www2.mmm.ucar.edu/wrf/users/wrf_users_guide/build/html/overview.html#initialization) might be helpful and answer some of those questions?

**Reply by jbensabat** [reply]
hi thanks a lot for your message
no worries , we are dealing with other problems these days.
the way I see it is that in order to create a model we need 4 files 
It could be good if there was a clear documentation about the structure of these files, what are the fields they can include and those they must include. 
I am ready to prepare such a document but I need info
I am aware of the real.exe  and geogrid. but they rely on data that is not relevant to us, as we have far more precise data hdf/netcdf format so we have created a software with a GUI that pulls the required data from various DBS.
best
jac

**Reply by aubreyd** [reply]
You do not need to use real.exe. We provide a utility script that can create the wrfinput file for you from your geogrid. We also have training materials that describe how to create the various input files needed.

https://github.com/NCAR/wrf_hydro_model_tools/tree/main/parameters/create_wrfinput
https://github.com/NCAR/wrf_hydro_training/tree/master/lessons/training
